### ISSUE TITLE
Compatibility Issue with Flutter 3.27.0: Compilation Fails Due to DialogTheme Assignment Errors

### ISSUE THREAD

**Issue by rignaneseleo** [nizwar/ndialog#16]
**Description:**

After upgrading to Flutter 3.27.0, I am unable to compile my project due to the following errors related to the `ndialog` package:

```
Launching lib/main.dart on Chrome in debug mode...
../../../../.pub-cache/hosted/pub.dev/ndialog-4.3.1/lib/src/ndialogBase.dart:33:50: Error: A value of type 'DialogThemeData' can't be assigned to a variable of type 'DialogTheme?'.
 - 'DialogThemeData' is from 'package:flutter/src/material/dialog_theme.dart' ('../../../../fvm/versions/stable/packages/flutter/lib/src/material/dialog_theme.dart').
 - 'DialogTheme' is from 'package:flutter/src/material/dialog_theme.dart' ('../../../../fvm/versions/stable/packages/flutter/lib/src/material/dialog_theme.dart').
    final DialogTheme? dialogTheme = DialogTheme.of(context);
                                                 ^
../../../../.pub-cache/hosted/pub.dev/ndialog-4.3.1/lib/src/progressDialog.dart:292:49: Error: A value of type 'DialogThemeData' can't be assigned to a variable of type 'DialogTheme'.
 - 'DialogThemeData' is from 'package:flutter/src/material/dialog_theme.dart' ('../../../../fvm/versions/stable/packages/flutter/lib/src/material/dialog_theme.dart').
 - 'DialogTheme' is from 'package:flutter/src/material/dialog_theme.dart' ('../../../../fvm/versions/stable/packages/flutter/lib/src/material/dialog_theme.dart').
    final DialogTheme dialogTheme = DialogTheme.of(context);
                                                ^
Failed to compile application.
```

It seems that the `DialogTheme.of(context)` method now returns a `DialogThemeData` type in Flutter 3.27.0, which causes a type mismatch when assigning it to variables of type `DialogTheme` or `DialogTheme?`.

**Steps to Reproduce:**
1. Upgrade Flutter to version 3.27.0.
2. Add the `ndialog` package (version 4.3.1) to your `pubspec.yaml`.
3. Try to compile a Flutter project that uses `ndialog`.

**Expected Behavior:**
The project should compile successfully without errors.

**Actual Behavior:**
The compilation fails due to type mismatch errors in the `ndialogBase.dart` and `progressDialog.dart` files.

**Proposed Solution:**
Update the `ndialog` package to account for the change in the return type of `DialogTheme.of(context)` in Flutter 3.27.0. Specifically:
- Ensure that `DialogThemeData` is correctly handled.
- Adjust type declarations where necessary to avoid assignment errors.

**Environment:**
- Flutter version: 3.27.0
- `ndialog` version: 4.3.1
- Platform: Chrome (Debug Mode)

Please let me know if I can assist further or provide additional details!

**Comment by rignaneseleo** [comment]
It looks like #15 is the fix.

**Comment by nizwar** [comment]
Fixes just published within v4.4.0, feel free to re-open it if you have similar concern
